ONSdigital/ras-rm-secure-message-v2
# RAS-298 Create POST message and POST thread

Updated by LJBabbage
# What and why?
This PR adds the endpoints to post a thread and a message. It also has implemented a simple sqlite database fixture structure and upped the tests (pytest framework) to 100% for the app (bar postgres which has been ignored as that can't be covered), visualisation of the tests has also been included. Mypy with type checking has been introduced and a tidy up of the app in general has been done (config files added, ignores files updated to reflect the repo, unused bits removed etc)

# How to test?

First you need a postgres db on port 5432, I regularly use the one from terminal trigger (ras trigger dev <namespace> / ras pf) but  that naturally will build out the other apps (N.b if doing that you will either have to knock out the old secure messaging on port 5051 or change this app to a different one) Once you have done that run this PR locally (make start) and grab an app that can generate post request, I use Postman but it can be anything. Now you can interact with the app try happy paths and unhappy paths, see images for end points, and tests for valid messages/threads

![Screenshot 2023-07-25 at 13 22 12](https://github.com/ONSdigital/ras-rm-secure-message-v2/assets/14179817/f865d9df-bb98-444d-bfd5-de8c0c17e1db)

![Screenshot 2023-07-25 at 13 23 05](https://github.com/ONSdigital/ras-rm-secure-message-v2/assets/14179817/d858325b-0fdd-456e-b6f4-c6896fe65eb1)

N.B  a message has to have a thread_id that exists to work, although this is naturally an unhappy path which is also unit-tested, that I changed my port to 5052, it could have been anything, and there are also no authentication headers at present
